hat/app/org/hatdex/hat/api/service/applications/ApplicationsService.scala

```diff
@@ -125,42 +128,91 @@ class ApplicationsService @Inject() (
+    val maybeDataDebitSetup: Option[Future[Done]] = for {
+      ddId <- application.application.dataDebitId
+      ddsRequest <- application.application.dataDebitSetupRequest
     } yield {
       logger.debug(s"Enable data debit $ddId for ${application.application.id}")
       for {
-        dds ← dataDebitService.createDataDebit(ddId, ddsRequest, user.userId)(hat)
+        _ <- dataDebitService.createDataDebit(ddId, ddsRequest, user.userId)(hat)
           .recoverWith({
             case _: RichDataDuplicateDebitException ⇒
               dataDebitService.updateDataDebitInfo(ddId, ddsRequest)(hat)
                 .flatMap(_ ⇒ dataDebitService.updateDataDebitPermissions(ddId, ddsRequest, user.userId)(hat))
           })
-        _ ← dataDebitService.dataDebitEnableNewestPermissions(ddId)(hat)
-      } yield dds
+        _ <- dataDebitService.dataDebitEnableNewestPermissions(ddId)(hat)
+      } yield Done
     }
 
-    // Set up the app
-    val appSetup = for {
-      _ ← maybeDataDebitSetup.getOrElse(Future.successful(())) // If data debit was there, must have been setup successfully
-      _ ← applicationSetupStatusUpdate(application, enabled = true)(hat.db) // Update status
+    maybeDataDebitSetup.getOrElse(Future.successful(Done)) // If data debit was there, must have been setup successfully
+  }
+
+  private def setupApplication(application: HatApplication)(implicit hat: HatServer, user: HatUser, requestHeader: RequestHeader): Future[HatApplication] = {
+    for {
+      _ <- applicationSetupStatusUpdate(application, enabled = true)(hat.db) // Update status
       _ <- statsReporter.registerOwnerConsent(application.application.id)
-      app ← applicationStatus(application.application.id, bustCache = true) // Refetch all information
-        .map(_.getOrElse(throw new RuntimeException("Application information not found during setup")))
+      app <- applicationStatus(application.application.id, bustCache = true) // Refetch all information
+        .map(_.getOrElse(throw HatApplicationSetupException(application.application.id, "Application information not found during setup")))
     } yield {
       logger.debug(s"Application status for ${app.application.id}: active = ${app.active}, setup = ${app.setup}, needs updating = ${app.needsUpdating}")
       app
     }
+  }
+
+  private def setupDependencies(application: HatApplication)(implicit hat: HatServer, user: HatUser, requestHeader: RequestHeader): Future[Option[Boolean]] = {
+    val maybeDependenciesSetup: Option[Future[Boolean]] = application.application.dependencies.map { deps =>
+      implicit val hatDb = hat.db
+
+      val appDeps = deps.plugs.toSet ++ deps.plugs.toSet
+      val toolDeps = deps.tools.toSet
+
+      val dependenciesEnabled = for {
+        appStatuses <- Future.sequence(appDeps.map(applicationStatus(_)))
+        updatedAppStatuses <- Future.sequence(appStatuses.map(_.get).filterNot(app => app.setup && app.enabled).map(setupApplication(_)))
```

Comment:
`.get` you cut some corners here didn't you?

Reply:
This is done purposefully - I wanted the error to be thrown in case any of the applications are not found. In previous commits `flatten` was used but that resulted in incorrect behaviour - failures were silently ignored.